packages/app-builder/src/components/Workflows/WorkflowAstProvider.tsx

```diff
@@ -0,0 +1,162 @@
+import { type AstNode, type DataModel } from '@app-builder/models';
+import { NewPayloadAstNode } from '@app-builder/models/astNode/data-accessor';
+import { NewNodeEvaluation } from '@app-builder/models/node-evaluation';
+import { generateFlatEvaluation } from '@app-builder/routes/ressources+/scenarios+/$scenarioId+/validate-ast';
+import { AstBuilderDataSharpFactory } from '@ast-builder/Provider';
+import { type ReactNode, useEffect, useMemo, useRef } from 'react';
+import { AstBuilderNodeSharpFactory } from '../AstBuilder/edition/node-store';
+
+interface WorkflowAstProviderProps {
+  children: ReactNode;
+  triggerObjectType: string;
+  dataModel: DataModel;
+  node: AstNode;
+  onChange: (node: AstNode) => void;
+}
+
+// Helper function to ensure node has proper structure for evaluation
+function normalizeAstNode(node: AstNode): AstNode {
+  return {
+    ...node,
+    children: node.children || [],
+    namedChildren: node.namedChildren || {},
+  };
+}
+
+export function WorkflowAstProvider({
+  children,
+  triggerObjectType,
+  dataModel,
+  node,
+  onChange,
+}: WorkflowAstProviderProps) {
+  // Create payload accessors from the trigger object fields
+  const payloadAccessors = useMemo(() => {
+    const triggerTable = dataModel.find((table) => table.name === triggerObjectType);
+    if (!triggerTable) {
+      return [];
+    }
+
+    return triggerTable.fields.map((field) => NewPayloadAstNode(field.name));
+  }, [dataModel, triggerObjectType]);
+
+  const dataSharp = AstBuilderDataSharpFactory.createSharp({
+    scenarioId: 'workflow',
+    data: {
+      triggerObjectType,
+      dataModel,
+      // Provide minimal data for workflow payload evaluation
+      databaseAccessors: [],
+      payloadAccessors,
+      customLists: [],
+    },
+    mode: 'edit',
+    showValues: false,
+  });
```

Comment:
Did you redefine all the internal working of ast builder data sharp to have an empty array in `databaseAccessors` ? I think it could have been avoided with a filter on the operand, what do you think?